ngraph/core/src/op/ctc_greedy_decoder_seq_len.cpp

```diff
@@ -0,0 +1,152 @@
+                                                       const bool merge_repeated,
+                                                       const element::Type& classes_index_type,
+                                                       const element::Type& sequence_length_type)
+    : Op({input, seq_len, blank_index})
+    , m_merge_repeated(merge_repeated)
+    , m_classes_index_type(classes_index_type)
+    , m_sequence_length_type(sequence_length_type)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v6::CTCGreedyDecoderSeqLen::validate_and_infer_types()
+{
+    const auto& logits_pshape = get_input_partial_shape(0);
+    const auto& seq_len_pshape = get_input_partial_shape(1);
+    auto input_et = get_input_element_type(0);
+
+    // output dynamic rank tensor if all inputs are of dynamic rank
+    if (logits_pshape.rank().is_dynamic() || seq_len_pshape.rank().is_dynamic())
+    {
+        set_output_type(
+            0, input_et, PartialShape{Dimension::dynamic(), Dimension::dynamic(), 1, 1});
+    }
+
+    // check ranks of input tensors
+    if (logits_pshape.rank().is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              logits_pshape.rank().get_length() == 3,
+                              "The rank of logits tensor must be equal to 3.");
+    }
+    if (seq_len_pshape.rank().is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              seq_len_pshape.rank().get_length() == 1,
+                              "The rank of sequence len tensor must be equal to 1.");
+    }
+
+    // check optional input type: blank index
+    if (get_input_size() == 3)
+    {
+        const auto& blank_index_type = get_input_element_type(2);
+        NODE_VALIDATION_CHECK(this,
+                              blank_index_type.is_integral_number(),
+                              "The blank index type is expected to be an integer type. Got: ",
+                              blank_index_type);
+    }
+
+    // validate input shapes and compute output shape
+    ngraph::Dimension batch_size = Dimension::dynamic();
+    ngraph::Dimension time_size = Dimension::dynamic();
+    if (logits_pshape.rank().is_static())
+    {
+        if (logits_pshape[0].is_static())
+        {
+            time_size = logits_pshape[0];
+        }
+        if (logits_pshape[1].is_static())
+        {
+            batch_size = logits_pshape[1];
+        }
+    }
+    if (seq_len_pshape.rank().is_static())
+    {
+        if (seq_len_pshape[0].is_static())
+        {
+            if (batch_size != Dimension::dynamic())
+            {
+                NODE_VALIDATION_CHECK(this,
+                                      seq_len_pshape[1] == batch_size,
+                                      "The first dimensions of input tensors must match.");
+            }
+            batch_size = seq_len_pshape[1];
+        }
+    }
+    set_output_type(0, input_et, PartialShape{batch_size, time_size});
```

Comment:
you missed use of interval logic, for example, for `batch_size`. You can compute more precise interval for output `batch_size` by interception of intervals from `logits` and `seq_len`